pkg/checks/dnsStatus/dnsStatus.go

```diff
@@ -0,0 +1,116 @@
+// Package dnsStatus implements a DNS checker for Kuberhealthy
+// It verifies that local DNS and external DNS are functioning correctly
+package dnsStatus
+
+import (
+	"errors"
+	"net"
+	"time"
+
+	// required for oidc kubectl testing
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+// Checker validates that DNS is functioning correctly
+type Checker struct {
+	FailureTimeStamp map[string]time.Time
+	Errors           []string
+	MaxTimeInFailure float64 // TODO - make configurable
+	client           *kubernetes.Clientset
+	Endpoints        []string // TODO - Make configurable
+}
+
+// New returns a new Checker
+func New(endpoints []string) *Checker {
+	defaultEndpoints := []string{
```

Comment:
While I think these are good places to resolve, we don't want everyone who uses kuberhealthy to get paged when aws or google goes offline (it should never happen though, right?!)

Perhaps use some root name server hostnames: https://www.iana.org/domains/root/servers ?

We could also make this do internal lookups by default only.  I am sure plenty of clusters don't actually resolve public DNS, or only are allowed to resolve a whitelist of external URLs... 

If we only resolve the internal dns, `kubernetes.default`, that should cover kubernetes.  If something is wrong upstream, then that kind of feels like its out of scope for what we monitor - a different DNS checker for external URLs could be run in the cluster.  I mean to say - we should probably _only_ guarantee that the kubernetes functionality passes.  We probably don't want to consider external resolution at the same time by default...